xls/ir/package.cc

```diff
@@ -181,6 +181,34 @@ std::string Package::SourceLocationToString(const SourceLocation loc) {
   return absl::StrFormat("%s:%d", filename, loc.lineno().value());
 }
 
+absl::StatusOr<Type*> Package::GetType(const Type* type) {
```

Comment:
How about checking that type is not owned by this package?

Reply:
Do you mean short circuit and return the type if it is already owned or return an error if it is already owened?  I prefer the former.

Reply:
To elaborate, the use case for function that I imagine is "I have a type of unknown provenance, give me a version that I know belongs to this package".  This seems more useful than making the caller check which package owns the type and crashing if they messed up.

Reply:
Sounds fine. The way things are currently implemented, mixing packages exposes a lot of foot guns around types. Probably want to have a singleton type manager so there is no potential for accidentally using a type in the wrong context.